lib/active_merchant/billing/gateways/braintree_blue.rb

```diff
@@ -690,7 +690,12 @@ def create_transaction_parameters(money, credit_card_or_vault_id, options)
           }
         end
 
-        parameters[:payment_method_nonce] = options[:three_ds_token] if options[:three_ds]
+        if options[:three_ds]
+          parameters[:payment_method_nonce] = options[:payment_method_nonce] if parameters[:payment_method_nonce]
+          parameters[:payment_method_nonce] = options[:three_ds_token] if parameters[:three_ds_token]
```

Comment:
Hmm, why do we have 2 assigns here?
Could we extract this logic into a separate method and then call it here to assign its result into `parameters[:payment_method_nonce]`?

Reply:
I refactored it.